common/src/main/java/org/opensearch/ml/common/ToolMetadata.java

```diff
@@ -70,6 +84,9 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
             builder.field(TOOL_TYPE_FIELD, type);
         }
         builder.field(TOOL_VERSION_FIELD, version != null ? version : "undefined");
+        if (attributes != null) {
```

Comment:
Attributes is optional, why do we use `protected static final Map<String, Object> DEFAULT_ATTRIBUTES = Collections.emptyMap();` rather than null?

Reply:
I think we should not return attributes if it's null or empty. That way we can keep the response clean